.github/workflows/main.yml

```diff
@@ -41,16 +41,27 @@ jobs:
 
     strategy:
       matrix:
+        os: [ubuntu-latest, macos-latest]
         python-version: ["3.7", "3.10"]
+        include:
+          - os: ubuntu-latest
+            os-installer: sudo apt-get
+            os-update: update
+            os-dependencies: install -y libopenmpi-dev openmpi-bin libhdf5-serial-dev
+          - os: macos-latest
+            os-installer: brew
+            os-update: update-reset
+            os-dependencies: install open-mpi
+
+    runs-on: ${{ matrix.os }}
 
-    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
 
-    - name: Install apt dependencies
+    - name: Install apt/brew dependencies
       run: |
-        sudo apt-get update
-        sudo apt-get install -y libopenmpi-dev openmpi-bin libhdf5-serial-dev
+        ${{ matrix.os-installer }} ${{ matrix.os-update }}
+        ${{ matrix.os-installer }} ${{ matrix.os-dependencies }}
```

Comment:
I think this would be easier to follow and customise if rather than having a single install step with the matrix customisations you just had two steps (Install apt; Install brew) which ran conditionally on each OS.

On macOS I wouldn't bother with the update-reset step which you would be able to skip if you broke it up